diff --git a/worker/meson.build b/worker/meson.build
index 45c5528f..a37c04f8 100644
--- a/worker/meson.build
+++ b/worker/meson.build
@@ -220,6 +220,18 @@ catch2_proj = subproject(
 libwebrtc_include_directories = include_directories('include')
 subdir('deps/libwebrtc')
 
+cpparg = custom_target(
+  'cpparg',
+  output : 'ms_cpparg.txt',
+  env: {
+    'CPPARGS': cpp_args,
+  },
+  command : [find_program('bash'), '-c', 'echo "${CPPARGS}" > ' + join_paths(meson.source_root(),'out','@OUTPUT@')],
+)
+cpparg_dep = declare_dependency(
+  sources: cpparg
+)
+
 dependencies = [
   abseil_cpp_proj.get_variable('absl_container_dep'),
   openssl_proj.get_variable('openssl_dep'),
@@ -228,6 +240,7 @@ dependencies = [
   libsrtp2_proj.get_variable('libsrtp2_dep'),
   usrsctp_proj.get_variable('usrsctp_dep'),
   libwebrtc_dep,
+  cpparg_dep,
 ]
 
 link_whole = [
